core/src/main/java/org/lflang/federated/generator/FederateInstance.java

```diff
@@ -698,24 +719,69 @@ public TimeValue findNearestPhysicalActionTrigger(ReactionInstance reaction) {
           // Logical action
           // Follow it upstream inside the reactor
           for (ReactionInstance uReaction : actionInstance.getDependsOnReactions()) {
-            // Avoid a loop
-            if (!Objects.equal(uReaction, reaction)) {
+            // Avoid a potentially deep loop by checking the visited set.
+            if (!visited.contains(uReaction)) {
               TimeValue uMinDelay =
-                  actionInstance.getMinDelay().add(findNearestPhysicalActionTrigger(uReaction));
+                  actionInstance
+                      .getMinDelay()
```

Comment:
This doesn't look quite right because this minDelay of this action will get ignored if the uReaction has been previously visited via some other path.

Reply:
This is resolved in 083c7081c9d1bebd6a815ff105eb24a3f616f21e